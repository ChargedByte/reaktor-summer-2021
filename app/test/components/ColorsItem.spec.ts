import { createLocalVue, mount } from '@vue/test-utils'

import Vuetify from 'vuetify'

import ColorsItem from '~/components/ColorsItem.vue'

describe('ColorsItem', () => {
  const localVue = createLocalVue()
  let vuetify: Vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('is a Vue instance', () => {
    const wrapper = mount(ColorsItem, {
      localVue,
      vuetify,
      propsData: { value: [] },
    })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly: empty', () => {
    const wrapper = mount(ColorsItem, {
      localVue,
      vuetify,
      propsData: { value: [] },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: Red', () => {
    const wrapper = mount(ColorsItem, {
      localVue,
      vuetify,
      propsData: { value: ['red'] },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: Red, Green', () => {
    const wrapper = mount(ColorsItem, {
      localVue,
      vuetify,
      propsData: { value: ['red', 'Green'] },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
