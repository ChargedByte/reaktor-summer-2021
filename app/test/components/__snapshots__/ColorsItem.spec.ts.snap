// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorsItem renders correctly: Red 1`] = `"<span>Red</span>"`;

exports[`ColorsItem renders correctly: Red, Green 1`] = `"<span>Red, Green</span>"`;

exports[`ColorsItem renders correctly: empty 1`] = `"<span></span>"`;
